python/pyarrow/parquet.py

```diff
@@ -1237,6 +1264,8 @@ def validate_schemas(self):
             if self.common_metadata is not None:
                 self.schema = self.common_metadata.schema
             else:
+                if not self.pieces:
+                    raise ValueError('No partition is found.')
```

Comment:
Is this a related change? Why was it needed?

Reply:
If no partition satisfies the filters, the next line will throw IndexOutOfRange error:
                self.schema = self.pieces[0].get_metadata().schema
I add the two lines to make the error more meaningful.

Reply:
Is there a test that covers this?

Reply:
I have remove this change from this JIRA/pull request. I will create another pull request for this particular issue.